src/NServiceBus.Msmq.AcceptanceTests/When_distributing_a_command.cs

```diff
@@ -11,103 +11,178 @@
-                    c.UnicastRouting().RouteToEndpoint(typeof(Request), ReceiverEndpoint);
+                    c.UnicastRouting().RouteToEndpoint(typeof(RequestA), ReceiverAEndpoint);
+                    c.UnicastRouting().RouteToEndpoint(typeof(RequestB), ReceiverBEndpoint);
                 });
             }
 
-            public class ResponseHandler : IHandleMessages<Response>
+            public class ResponseHandler :
+                IHandleMessages<ResponseA>,
+                IHandleMessages<ResponseB>
             {
                 public Context Context { get; set; }
 
-                public Task Handle(Response message, IMessageHandlerContext context)
+                public Task Handle(ResponseA message, IMessageHandlerContext context)
+                {
+                    switch (message.EndpointName)
+                    {
+                        case "ReceiverA1":
+                            Context.ReceiverA1TimesCalled++;
+                            break;
+                        case "ReceiverA2":
+                            Context.ReceiverA2TimesCalled++;
+                            break;
+                    }
+
+                    return context.Send(new RequestB());
+                }
+
+                public Task Handle(ResponseB message, IMessageHandlerContext context)
                 {
                     switch (message.EndpointName)
                     {
-                        case "Receiver1":
-                            Context.Receiver1TimesCalled++;
+                        case "ReceiverB1":
+                            Context.ReceiverB1TimesCalled++;
                             break;
-                        case "Receiver2":
-                            Context.Receiver2TimesCalled++;
+                        case "ReceiverB2":
+                            Context.ReceiverB2TimesCalled++;
                             break;
                     }
 
-                    return context.Send(new Request());
+                    Context.MessagesReceivedPerEndpoint++;
+                    if (Context.MessagesReceivedPerEndpoint < numberOfMessagesToSendPerEndpoint)
+                    {
+                        return context.Send(new RequestA());
+                    }
+
+                    return Task.CompletedTask;
+                }
+            }
+        }
+
+        public class ReceiverA : EndpointConfigurationBuilder
+        {
+            public ReceiverA()
+            {
+                EndpointSetup<DefaultServer>();
+            }
+
+            public class MyMessageHandler : IHandleMessages<RequestA>
+            {
+                public ReadOnlySettings Settings { get; set; }
+
+                public Task Handle(RequestA message, IMessageHandlerContext context)
+                {
+                    return context.Reply(new ResponseA
+                    {
+                        EndpointName = Settings.Get<string>("Name")
```

Comment:
as described above, I don't think this would work because the endpoint name does not contain any instance information.